### PR TITLE
Refactoring workflows for RPM support + various workflow fixes

### PR DESCRIPTION
**Pull request by AhmedBM**
## Description

* Various workflow improvements (package-type support)
* Added `ubuntu 22.04` to e2e tests
* Removed `ubuntu 18.04` from build/test/publish workflows
* Fixed `CommandRunner` and `DeliveryOptimization` test recipes to work in azurelinux (bash v5)
* Updated actions to latest versions (fixed Node JS 1.6 warning)

## Checklist

- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: .github/workflows/prod.yml [hunk 1]
```diff
@@ -13,7 +13,6 @@ jobs:
       matrix:
         target:
           [
-            { os: ubuntu, version: 18.04, dist: bionic, package-type: DEB },
```
Comment (robertschaedler3) [inline .github/workflows/prod.yml:16]: Does this mean we are officially dropping support for Ubuntu 18.04 packages?

**Review by robertschaedler3 — APPROVED** [review]

File: .github/workflows/prod.yml (hunk 1 above)
Comment (robertschaedler3) [reply .github/workflows/prod.yml:16]: If so, we need to update the required checks in our branch protection rules for `main` to remove the 2 Ubuntu 18.04 checks that aren't running anymore and add the new Mariner/Ubuntu 22.04 ones.

File: .github/workflows/module-test.yml [hunk 2]
```diff
@@ -9,8 +9,17 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        target: [ubuntu-18.04, ubuntu-20.04, ubuntu-22.04, debian-10, debian-11]
+        target:
+          [
+            { os: ubuntu, version: 20.04, package-type: DEB, excluded-tests: "SecurityBaseline|Sample" },
```
Comment (simonjaeger) [inline .github/workflows/module-test.yml:14]: Why is SecurityBaseline excluded?

**Review by simonjaeger — APPROVED** [review]

File: .github/workflows/module-test.yml (hunk 2 above)
Comment (AhmedBM) [reply .github/workflows/module-test.yml:14]: Its broken and fails on every distro. Its been excluded for now.

Comment (simonjaeger) [reply .github/workflows/module-test.yml:14]: Is this being tracked? What is required to fix it?

Comment (AhmedBM) [reply .github/workflows/module-test.yml:14]: Yes, this will be part of my summary email to the team about the current state/gaps, etc.

**Pull request merged**